Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -270,75 +270,99 @@ void updateAncestorNodesAfterDelete(NodeReference nodeReference, Point point) {
         updateAncestorPointSum(nodeReference);
     }
 
+    NodeReference findLeaf(Point point, boolean deleteMass) {
+        NodeReference nodeReference = root;
+        while (!isLeaf(nodeReference)) {
+            if (deleteMass) {
+                decrementMass(nodeReference);
+            }
+            nodeReference = (leftOf(point, getCutDimension(nodeReference), getCutValue(nodeReference)))
+                    ? getLeftChild(nodeReference)
+                    : getRightChild(nodeReference);
+        }
+        return nodeReference;
+    }
+
+    NodeReference findLeafAndVerify(Point point, boolean deleteMass) {
+        NodeReference nodeReference = findLeaf(point, deleteMass);
+        Point oldPoint = getPointFromLeafNode(nodeReference);
+        if (!equals(oldPoint, point)) {
+            throw new IllegalStateException(toString(point) + " " + toString(getPointFromLeafNode(nodeReference)) + " "
```

Comment:
Won't this leave the tree in an invalid state if `deleteMass` is true?

Reply:
Not in the current setup - but someone may wish to invoke deletePoint when the point is not in the tree. Changed.